_ml-commons-plugin/api.md

```diff
@@ -0,0 +1,740 @@
+          "_id" : "-QkKJX8BvytMh9aUeuLD",
+          "_version" : 1,
+          "_seq_no" : 12,
+          "_primary_term" : 15,
+          "_score" : 2.4159138,
+          "_source" : {
+            "name" : "FIT_RCF",
+            "version" : 1,
+            "content" : "xxx",
+            "algorithm" : "FIT_RCF"
+          }
+        },
+        {
+          "_index" : ".plugins-ml-model",
+          "_type" : "_doc",
+          "_id" : "OxkvHn8BNJ65KnIpck8x",
+          "_version" : 1,
+          "_seq_no" : 2,
+          "_primary_term" : 8,
+          "_score" : 2.4159138,
+          "_source" : {
+            "name" : "FIT_RCF",
+            "version" : 1,
+            "content" : "xxx",
+            "algorithm" : "FIT_RCF"
+          }
+        }
+      ]
+    }
+  }
+```
+
+## Delete model
+
+Deletes a model based on the model_id
+
+```json
+DELETE /_plugins/_ml/models/<model_id>
+```
+
+The API returns the following:
+
+```json
+{
+  "_index" : ".plugins-ml-model",
+  "_type" : "_doc",
+  "_id" : "MzcIJX8BA7mbufL6DOwl",
+  "_version" : 2,
+  "result" : "deleted",
+  "_shards" : {
+    "total" : 2,
+    "successful" : 2,
+    "failed" : 0
+  },
+  "_seq_no" : 27,
+  "_primary_term" : 18
+}
+```
+
+## Predict
+
+ML commons can predict new data with your trained model either from indexed data or a data frame. The model_id is required to use the Predict API.
+
+```json
+POST /_plugins/_ml/_predict/<algorithm_name>/<model_id>
+```
+
+### Request
+
+```json
+POST /_plugins/_ml/_predict/kmeans/<model-id>
+{
+    "input_query": {
+        "_source": ["petal_length_in_cm", "petal_width_in_cm"],
+        "size": 10000
+    },
+    "input_index": [
+        "iris_data"
+    ]
+}
+```
+
+### Response
+
+```json
+{
+  "status" : "COMPLETED",
+  "prediction_result" : {
+    "column_metas" : [
+      {
+        "name" : "ClusterID",
+        "column_type" : "INTEGER"
+      }
+    ],
+    "rows" : [
+      {
+        "values" : [
+          {
+            "column_type" : "INTEGER",
+            "value" : 1
+          }
+        ]
+      },
+      {
+        "values" : [
+          {
+            "column_type" : "INTEGER",
+            "value" : 1
+          }
+        ]
+      },
+      {
+        "values" : [
+          {
+            "column_type" : "INTEGER",
+            "value" : 0
+          }
+        ]
+      },
+      {
+        "values" : [
+          {
+            "column_type" : "INTEGER",
+            "value" : 0
+          }
+        ]
+      },
+      {
+        "values" : [
+          {
+            "column_type" : "INTEGER",
+            "value" : 0
+          }
+        ]
+      },
+      {
+        "values" : [
+          {
+            "column_type" : "INTEGER",
+            "value" : 0
+          }
+        ]
+      }
+    ]
+  }
+```
+
+
+## Train and Predict
+
+Use to train and then immediately predict against the same training data set. Can only be used with unsupervised learning models and the following algorithms:
```

Comment:
"Can only be used with unsupervised learning models and the following algorithms:" -> "Can only be used for unsupervised learning algorithms and now we support three algorithms:"

Reply:
The first is grammatically correct. How about, 

"Can only be used with the following unsupervised learning algorithms?"

Reply:
Sure, that sounds good for ""Can only be used with the following unsupervised learning algorithms?""